lib/application.js

```diff
@@ -471,15 +539,47 @@ var RequireAuthorizationApplication = Application.extend({
             });
         });
     },
-    getAccessToken: function(token, secret, verifier, callback, options) {
+    setAccessToken: function(token, secret, verifier, callback, options) {
         var self = this;
 
         return new Promise(function(resolve, reject) {
-            self.oa.getOAuthAccessToken(token, secret, verifier, function(err, token, secret, results) {
+            self.oa.getOAuthAccessToken(token, secret, verifier, function(err, results) {
                 if (err)
                     reject(err);
-                else
-                    resolve({ token: token, secret: secret, results: results });
+                else {
+                    var exp = new Date();
+                    exp.setTime(exp.getTime() + (results.oauth_expires_in * 1000));
+                    self.setOptions({
+                        accessToken: results.oauth_token,
+                        accessSecret: results.oauth_token_secret,
+                        sessionHandle: results.oauth_session_handle,
+                        tokenExpiry: exp.toString()
+                    });
+                    resolve({ results: results });
```

Comment:
Once we've refreshed the token, how does it get back out to the user? It would be best for us to pass the token back out so that it can be persisted to disk ready for the next run of the app.

I think the neatest way to do this would be to emit a `token_refreshed` event or something similar.